action.yml

```diff
@@ -0,0 +1,29 @@
+name: 'xrefcheck'
+description: 'Check cross-references in your repository'
+author: 'Serokell'
+inputs:
+  xrefcheck-version:
+    required: false
+    description: 'Version of xrefcheck to use. If set to "latest", it will always get the latest stable version.'
+    default: 'latest'
+  xrefcheck-args:
+    required: false
+    description: 'Arguments passed to xrefcheck (verbatim)'
+  action:
+    required: false
+    description: |
+      Action you want to be performed, can be one of:
+      * install – just install the tool, its path will be in this action's output.
```

Comment:
When only install can be useful? The executable will lie within the container and you won't be able to run it from outside anyway, unless you later run the same container and make sure in advance that it has a volume attached to persist data.

Reply:
Let me check that. If it doesn't work, I guess I'll just remove this option.

Reply:
@Martoon-00 I don't really know how this stuff in Docker works under the hood, but in this particular case `install` seems to work just fine as can be seen here: https://github.com/serokell/universum/pull/224/checks?check_run_id=801204501

I am specifying `action: install` and after that `bin/xrefcheck --help` works fine there. Note that one should use `xrefcheck-path` there instead of `bin/xrefcheck`, I used it because it's simpler (but not proper).

> When only install can be useful?

I think in normal scenario you just want to use `action: run` and that's the default which should suffice for nearly everyone. However, it's hard to consider all possible usages one may ever need. I can imagine that maybe someone will want to run the tool multiple times with different options for example :man_shrugging: 

That being said, it indeed might be that the way I implement `install` is not the best one and not proper. In particular, it's probably better to put this executable into some standard global path, perhaps one from `$PATH`. The fact that `xrefcheck-path` is a relative path looks somewhat fragile.

I see three options:
1. Throw-away this `action` parameter completely and always run `xrefcheck`.
2. Keep things as they are, write that `action: install` is experimental functionality (as well as `xrefcheck-path`) that was tested and works, but might need to be implemented differently.
3. Investigate how other (popular) GH Actions install software and implement it the same way.

The third option is the best one, but sadly I don't have much time to work on this tool and this option may delay merge of this PR significantly. But I'd like to merge it more or less ASAP and mention this action in `xrefcheck` README (and then finally start advertising it). So, if you don't mind, I will choose the second option.

Reply:
So what about this one?

Reply:
Aha okay. Looks like this works because github actions automatically mount the project repository into docker container, so the executable is accessible outside of container.

I also thought that each execution of xrefcheck would require a separate build step, but now taking a look at `Checks` tab I see that build executes only once, and then there ar multiple run steps. So install apparently makes sense.

Let's leave `install` option then.